### PR TITLE
Progress in Collections is no longer lost upon server shutdown #326

### LINKED ISSUE #326: Collections do not get stored permanently
Click/Activate some collections
Check collections
Shut down server
Boot up server
Check collections

All set to 0 again

### PR DESCRIPTION
**Pull request by madsboddum**
Fixes #326

Unable to test, getting compilation errors in ProjectSWGRunner.

### CODE REVIEW

File: src/main/java/com/projectswg/holocore/resources/support/objects/swg/player/PlayerObjectShared.java [hunk 1]
```diff
@@ -390,6 +392,8 @@ public void readMongo(MongoData data) {
 		lifetimeGcwPoints = data.getLong("lifetimeGcwPoints", lifetimeGcwPoints);
 		showBackpack = data.getBoolean("showBackpack", showBackpack);
 		showHelmet = data.getBoolean("showHelmet", showHelmet);
+		collectionBadges.xor(BitSet.valueOf(data.getByteArray("collectionBadges", new byte[]{})));
+		collectionBadges2.xor(BitSet.valueOf(data.getByteArray("collectionBadges2", new byte[]{})));
```
Comment (Josh-Larson) [inline src/main/java/com/projectswg/holocore/resources/support/objects/swg/player/PlayerObjectShared.java:396]: I think this assumes that the collectionBadges collections are empty. Can you explicitly clear them both just prior to this command? Then you can just use .or()

Comment (madsboddum) [reply src/main/java/com/projectswg/holocore/resources/support/objects/swg/player/PlayerObjectShared.java]: Good idea. I found some code elsewhere that did the same thing and tried to reduce duplication a bit.

**Review by Josh-Larson — APPROVED** [review]
Great refactor!

**Comment by Undercova** [comment]
Tested and works fine
Good job !

**Pull request merged**